src/StreamSelectLoop.php

```diff
@@ -252,7 +250,10 @@ protected function streamSelect(array &$read, array &$write, $timeout)
         if ($read || $write) {
             $except = null;
 
-            return stream_select($read, $write, $except, $timeout === null ? null : 0, $timeout);
+            $tv_sec = $timeout === null ?: floor($timeout);
+            // convert the fractional part of $timeout to microseconds
+            $tv_usec = $timeout === null ?: round(fmod($timeout, 1) * self::MICROSECONDS_PER_SECOND);
```

Comment:
If `$timeout === null`, both `$tv_sec` and `$tv_usec` will become `true` because of the elvis operator.

This should be
```php
$tv_sec = $timeout === null ? null : floor($timeout);
// convert the fractional part of $timeout to microseconds
$tv_usec = $timeout === null ? null : round(fmod($timeout, 1) * self::MICROSECONDS_PER_SECOND);
```